src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -193,68 +195,95 @@ public void transform(FeedTransformZipTarget zipTarget, MonitorableJob.Status st
-                        transformedValue = convertToTitleCase(transformedValue);
-                    }
-                    // TODO: Implement other capitalization styles.
-                }
-
-                // Perform substitutions if any.
-                transformedValue = performSubstitutions(transformedValue);
-
-                // Re-assemble the CSV line and place in buffer.
-                String[] csvValues = csvReader.getValues();
-                csvValues[transformFieldIndex] = transformedValue;
-
-                // Write line to table (plus new line char).
-                writer.write(csvValues);
+            int modifiedRowCount = generateCsvContent(writer, headers, csvReader, transformFieldIndex);
 
-                // Count number of CSV rows changed.
-                if (!originalValue.equals(transformedValue)) {
-                    modifiedRowCount++;
-                }
-            } // End of iteration over each row.
-            csvReader.close();
-            writer.flush();
-
-            // Copy csv input stream into the zip file, replacing the existing file.
-            try (
-                // Modify target zip file that we just read.
-                FileSystem targetZipFs = FileSystems.newFileSystem(tempZipPath, (ClassLoader) null);
-                // Stream for file copy operation.
-                InputStream inputStream =  new ByteArrayInputStream(stringWriter.toString().getBytes(StandardCharsets.UTF_8))
-            ) {
-                Path targetTxtFilePath = getTablePathInZip(tableName, targetZipFs);
-                Files.copy(inputStream, targetTxtFilePath, StandardCopyOption.REPLACE_EXISTING);
-                zipTarget.feedTransformResult.tableTransformResults.add(
-                    new TableTransformResult(tableName, 0, modifiedRowCount, 0)
-                );
-            }
+            writeCsvContent(zipTarget, tempZipPath, stringWriter, tableName, modifiedRowCount);
 
             // Replace original zip file with temporary working zip file.
             // (This should also trigger a system IO update event, so subsequent IO calls pick up the correct file.
             Files.move(tempZipPath, originalZipPath, StandardCopyOption.REPLACE_EXISTING);
             LOG.info("Field normalization transformation successful, {} row(s) changed.", modifiedRowCount);
+        } catch (ZipException ze) {
+            status.fail(
+                String.format("'Normalize Field' failed because the GTFS archive is corrupted (%s).", ze.getMessage()),
+                ze
+            );
         } catch (Exception e) {
             status.fail("Unknown error encountered while transforming zip file", e);
         }
     }
 
+    /** Write csv input stream into the zip file, replacing the existing file. */
+    private void writeCsvContent(FeedTransformZipTarget zipTarget, Path tempZipPath, StringWriter stringWriter, String tableName, int modifiedRowCount) throws IOException {
```

Comment:
Nit. Wrap line.

Reply:
Updated in https://github.com/ibi-group/datatools-server/pull/598/commits/c50f9a66d66d4304addbbf0b59c3b154cbb732f5